src/shared/condition.c

```diff
@@ -138,6 +140,49 @@ static int condition_test_kernel_command_line(Condition *c) {
         return false;
 }
 
+static int condition_test_user(Condition *c) {
+        uid_t id;
+        int r;
+        _cleanup_free_ const char * username = NULL;
```

Comment:
does this even compile? pointers that are eventually passed to `free()` can't be "const", as freeing means changes to memory. (also see the free() function signature, it lacks "const" on the parameter). Hence, please drop "const" here...